cli/cmd/proxy/loglevel/command.go

```diff
@@ -0,0 +1,318 @@
+	"github.com/posener/complete"
+	helmCLI "helm.sh/helm/v3/pkg/cli"
+	"k8s.io/apimachinery/pkg/api/validation"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	defaultAdminPort    = 19000
+	flagNameNamespace   = "namespace"
+	flagNameKubeConfig  = "kubeconfig"
+	flagNameKubeContext = "context"
+)
+
+type LoggerConfig map[string]string
+
+var ErrMissingPodName = errors.New("Exactly one positional argument is required: <pod-name>")
+
+var levelToColor = map[string]string{
+	"trace":    terminal.Green,
+	"debug":    terminal.HiWhite,
+	"info":     terminal.Blue,
+	"warning":  terminal.Yellow,
+	"error":    terminal.Red,
+	"critical": terminal.Magenta,
+	"off":      "",
+}
+
+type LogLevelCommand struct {
+	*common.BaseCommand
+
+	kubernetes kubernetes.Interface
+	set        *flag.Sets
+
+	// Command Flags
+	podName     string
+	namespace   string
+	kubeConfig  string
+	kubeContext string
+
+	once            sync.Once
+	help            string
+	restConfig      *rest.Config
+	logLevelFetcher func(context.Context, common.PortForwarder) (LoggerConfig, error)
+}
+
+func (l *LogLevelCommand) init() {
+	l.set = flag.NewSets()
+	f := l.set.NewSet("Command Options")
+	f.StringVar(&flag.StringVar{
+		Name:    flagNameNamespace,
+		Target:  &l.namespace,
+		Usage:   "The namespace where the target Pod can be found.",
+		Aliases: []string{"n"},
+	})
+
+	f = l.set.NewSet("Global Options")
+	f.StringVar(&flag.StringVar{
+		Name:    flagNameKubeConfig,
+		Aliases: []string{"c"},
+		Target:  &l.kubeConfig,
+		Usage:   "Set the path to kubeconfig file.",
+	})
+	f.StringVar(&flag.StringVar{
+		Name:   flagNameKubeContext,
+		Target: &l.kubeContext,
+		Usage:  "Set the Kubernetes context to use.",
+	})
+
+	l.help = l.set.Help()
+}
+
+func (l *LogLevelCommand) Run(args []string) int {
+	l.once.Do(l.init)
+	l.Log.ResetNamed("loglevel")
+	defer common.CloseWithError(l.BaseCommand)
+
+	err := l.parseFlags(args)
+	if err != nil {
+		l.UI.Output(err.Error(), terminal.WithErrorStyle())
+		l.UI.Output(fmt.Sprintf("\n%s", l.Help()))
+		return 1
+	}
+
+	err = l.validateFlags()
+	if err != nil {
+		l.UI.Output(err.Error(), terminal.WithErrorStyle())
+		l.UI.Output(fmt.Sprintf("\n%s", l.Help()))
+		return 1
+	}
+
+	if l.logLevelFetcher == nil {
+		l.logLevelFetcher = FetchLogLevel
+	}
+
+	err = l.initKubernetes()
+	if err != nil {
+		l.UI.Output(err.Error(), terminal.WithErrorStyle())
+		l.UI.Output(fmt.Sprintf("\n%s", l.Help()))
+		return 1
+	}
+
+	adminPorts, err := l.fetchAdminPorts()
+	if err != nil {
+		l.UI.Output(err.Error(), terminal.WithErrorStyle())
+		l.UI.Output(fmt.Sprintf("\n%s", l.Help()))
+		return 1
+	}
+
+	logLevels, err := l.fetchLogLevels(adminPorts)
+	if err != nil {
+		l.UI.Output(err.Error(), terminal.WithErrorStyle())
+		l.UI.Output(fmt.Sprintf("\n%s", l.Help()))
+		return 1
+	}
+	l.outputLevels(logLevels)
+	return 0
+}
+
+func (l *LogLevelCommand) parseFlags(args []string) error {
```

Comment:
no reason in particular, this definitely reads a lot clearer

Reply:
so the flag lib doesn't actually handle parsing keyword arguments in this situation because it stops parsing args once it sees the first positional argument:
> Flag parsing stops just before the first non-flag argument ("-" is a non-flag argument) or after the terminator "--".
from:
https://pkg.go.dev/flag#hdr-Command_line_flag_syntax

so we have to pull all the positional arguments out of the args slice before calling parse on them